### PR TITLE
feature/hardware-wallets [v5]

### PR DESCRIPTION
**Pull request by gamalielhere**
### Feature

- [x] Updated CHANGELOG.md
- [x] Add PR label
- [x] No hard-coded strings

### CODE REVIEW

File: src/layouts/InterfaceLayout/InterfaceLayout.vue [hunk 1]
```diff
@@ -354,6 +364,20 @@ export default {
             })
             .catch(KeepkeyWallet.errorHandler);
           break;
+        case BC_VAULT:
+          BCVaultWalletInstance.init()
```
Comment (kvhnuke) [inline src/layouts/InterfaceLayout/InterfaceLayout.vue]: why is this defined in both hardwaremodal and here?

File: src/layouts/AccessWalletLayout/AccessWalletLayout.vue [hunk 2]
```diff
@@ -1,22 +1,346 @@
...
+    },
+    finneyModalOpen() {
+      this.$refs.finney.$refs.finneyModal.show();
+    },
+    xwalletModalOpen() {
+      this.$refs.xwallet.$refs.xwalletModal.show();
+    },
+
+    hardwareWalletOpen(wallet) {
+      // if (this.$refs.mnemonicPhrasePassword.$refs.password.visible) {
```
Comment (kvhnuke) [inline src/layouts/AccessWalletLayout/AccessWalletLayout.vue]: commented code

File: src/layouts/InterfaceLayout/InterfaceLayout.vue [hunk 3]
```diff
@@ -155,11 +162,14 @@ import {
   BITBOX as BITBOX_TYPE,
   SECALOT as SECALOT_TYPE,
   KEEPKEY as KEEPKEY_TYPE,
-  MNEMONIC as MNEMONIC_TYPE
+  MNEMONIC as MNEMONIC_TYPE,
+  BCVAULT as BC_VAULT
 } from '@/wallets/bip44/walletTypes';
+const BCVaultWalletInstance = BCVaultWallet();
```
Comment (kvhnuke) [inline src/layouts/InterfaceLayout/InterfaceLayout.vue]: this should be defined when they select bcvault

File: src/wallets/hardware/bcvault/index.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        });
+        const newTx = {};
+        newTx['feeCount'] = new BigNumber(
+          bufferToHex(tx['gasLimit'])
+        ).toNumber();
+        newTx['feePrice'] = new BigNumber(
+          bufferToHex(tx['gasPrice'])
+        ).toString();
+        newTx['amount'] =
+          new BigNumber(bufferToHex(tx['value'])).toNumber() || 0;
```
Comment (kvhnuke) [inline src/wallets/hardware/bcvault/index.js]: this has to be toString has well, web3-utils has hexToNumber you can use all across

File: src/wallets/hardware/bcvault/index.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        newTx['contractData'] = bufferToHex(tx['data']);
+        newTx['to'] = bufferToHex(tx['to']);
+        newTx['from'] = this.selectedAddress;
```
Comment (kvhnuke) [inline src/wallets/hardware/bcvault/index.js]: you can use tx.from, instead of this.selectedAddress

File: src/wallets/hardware/bcvault/index.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        if (tx.hasOwnProperty('nonce')) {
```
Comment (kvhnuke) [inline src/wallets/hardware/bcvault/index.js]: nonce will be always there, this if statement can be reduced to 
`bufferToHex(tx['nonce']) === '0x' ?0 :  hexToNumber(tx['nonce']))`

File: src/wallets/hardware/bcvault/index.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+          .catch(errorHandler);
+        if (result) {
+          const resultTx = new Transaction(result);
+          tx.v = getBufferFromHex(sanitizeHex(resultTx.v.toString('hex')));
+          tx.r = getBufferFromHex(sanitizeHex(resultTx.r.toString('hex')));
+          tx.s = getBufferFromHex(sanitizeHex(resultTx.s.toString('hex')));
+          const signedChainId = calculateChainIdFromV(tx.v);
+          if (signedChainId !== networkId)
+            throw new Error(
+              'Invalid networkId signature returned. Expected: ' +
```
Comment (kvhnuke) [inline src/wallets/hardware/bcvault/index.js]: this should be passed through i18n

File: src/layouts/InterfaceLayout/InterfaceLayout.vue (hunk 1 above)
Comment (gamalielhere) [reply src/layouts/InterfaceLayout/InterfaceLayout.vue]: for the switch address

File: src/wallets/hardware/bcvault/index.js (hunk 7 above)
Comment (gamalielhere) [reply src/wallets/hardware/bcvault/index.js]: This is fixed in coolwallet branch #2087

File: src/layouts/AccessWalletLayout/components/HardwareModal/HardwareModal.vue [hunk 8]
```diff
@@ -259,14 +290,60 @@ export default {
...
+                    } else {
+                      Toast.responseHandler(
+                        new Error('No wallet instance!'),
+                        Toast.ERROR
+                      );
+                    }
+                  })
+                  .catch(() => {
+                    Toast.responseHandler(
+                      new Error('Having issues with pairing cool wallet'),
```
Comment (gesseekur) [inline src/layouts/AccessWalletLayout/components/HardwareModal/HardwareModal.vue]: translations heh heh heh

File: src/main.js [hunk 9]
```diff
@@ -132,37 +132,39 @@ const vue = new Vue({
...
-      walletType: identifier
-    };
-    return new Promise(resolve => {
-      vue.$eventHub.$emit('issueModal', event, resolve);
-    }).then(res => {
-      return res === true ? event : null;
-    });
-  }
-});
+if (NODE_ENV === 'production') {
```
Comment (gesseekur) [inline src/main.js]: nicee!

File: src/layouts/AccessWalletLayout/components/HardwareModal/HardwareModal.vue (hunk 8 above)
Comment (kvhnuke) [reply src/layouts/AccessWalletLayout/components/HardwareModal/HardwareModal.vue]: yea where is the translations?

**Pull request merged**